src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -772,6 +772,77 @@ public bool IsTopLevelNodeWithMembers(SyntaxNode node)
+                    nameSuffix = GetTypeParameterSuffix(delegateDecl.TypeParameterList);
+                    containerDisplayName = GetContainerDisplayName(node.Parent);
+                    return;
+                case SyntaxKind.EnumDeclaration:
+                    var enumDecl = (EnumDeclarationSyntax)node;
+                    nameSuffix = "";
+                    containerDisplayName = GetContainerDisplayName(node.Parent);
+                    return;
+                case SyntaxKind.EnumMemberDeclaration:
+                    var enumMember = (EnumMemberDeclarationSyntax)node;
+                    nameSuffix = "";
+                    containerDisplayName = GetContainerDisplayName(node.Parent);
+                    return;
+                case SyntaxKind.EventDeclaration:
+                    var eventDecl = (EventDeclarationSyntax)node;
+                    nameSuffix = "";
+                    containerDisplayName = GetContainerDisplayName(node.Parent);
+                    return;
+                case SyntaxKind.IndexerDeclaration:
+                    var indexerDecl = (IndexerDeclarationSyntax)node;
+                    nameSuffix = GetIndexerSuffix(indexerDecl);
+                    containerDisplayName = GetContainerDisplayName(node.Parent);
+                    return;
+                case SyntaxKind.InterfaceDeclaration:
+                    var interfaceDecl = (InterfaceDeclarationSyntax)node;
+                    nameSuffix = GetTypeParameterSuffix(interfaceDecl.TypeParameterList);
+                    containerDisplayName = GetContainerDisplayName(node.Parent);
+                    return;
+                case SyntaxKind.MethodDeclaration:
+                    var method = (MethodDeclarationSyntax)node;
+                    nameSuffix = GetMethodSuffix(method);
+                    containerDisplayName = GetContainerDisplayName(node.Parent);
+                    return;
+                case SyntaxKind.PropertyDeclaration:
+                    var property = (PropertyDeclarationSyntax)node;
+                    nameSuffix = "";
+                    containerDisplayName = GetContainerDisplayName(node.Parent);
+                    return;
+                case SyntaxKind.StructDeclaration:
+                    var structDecl = (StructDeclarationSyntax)node;
+                    nameSuffix = GetTypeParameterSuffix(structDecl.TypeParameterList);
+                    containerDisplayName = GetContainerDisplayName(node.Parent);
+                    return;
+                case SyntaxKind.VariableDeclarator:
+                    // could either be part of a field declaration or an event field declaration
+                    var fieldDeclaration = (BaseFieldDeclarationSyntax)node.Parent.Parent;
+                    nameSuffix = "";
+                    containerDisplayName = GetContainerDisplayName(fieldDeclaration.Parent);
+                    return;
+            }
+
+            nameSuffix = "";
```

Comment:
Move into a `default` with a `return` so no case can forget to initialize and return?